mlflow/data/spark_dataset_source.py

```diff
@@ -0,0 +1,55 @@
+from typing import TypeVar, Any, Union, Optional, Mapping, Sequence, Dict
+
+from mlflow.data.dataset_source import DatasetSource
+from pyspark.sql import SparkSession, DataFrame
+
+
+SparkDatasetSourceType = TypeVar("SparkDatasetSourceType", bound="SparkDatasetSource")
+
+
+class SparkDatasetSource(DatasetSource):
+    def __init__(
+        self,
+        path: str,
```

Comment:
We should make sure to support spark table names, spark table paths, and spark SQL